avm/res/cache/redis-enterprise/database/main.bicep

```
@@ -0,0 +1,330 @@
+// ============== //
+
+resource redisCluster 'Microsoft.Cache/redisEnterprise@2024-09-01-preview' existing = {
+  name: redisClusterName
+}
+
+var clusterSku = redisCluster.sku.name
+var isAmr = startsWith(clusterSku, 'Balanced') || startsWith(clusterSku, 'ComputeOptimized') || startsWith(
+  clusterSku,
+  'FlashOptimized'
+) || startsWith(clusterSku, 'MemoryOptimized')
+
+resource redisDatabase 'Microsoft.Cache/redisEnterprise/databases@2024-09-01-preview' = {
+  parent: redisCluster
+  name: name
+  properties: {
+    accessKeysAuthentication: isAmr ? accessKeysAuthentication : null
+    clientProtocol: clientProtocol
+    clusteringPolicy: clusteringPolicy
+    deferUpgrade: deferUpgrade
+    evictionPolicy: evictionPolicy
+    geoReplication: !empty(geoReplication) ? geoReplication : null
+    modules: modules
+    port: port
+    persistence: persistence.type != 'disabled'
+      ? {
+          aofEnabled: persistence.type == 'aof'
+          rdbEnabled: persistence.type == 'rdb'
+          aofFrequency: persistence.type == 'aof' ? persistence.frequency : null
+          rdbFrequency: persistence.type == 'rdb' ? persistence.frequency : null
+        }
+      : null
+  }
+}
+
+module database_accessPolicyAssignments 'access-policy-assignment/main.bicep' = [
+  for (assignment, index) in (accessPolicyAssignments ?? []): {
+    name: '${uniqueString(deployment().name)}-redis-apa-${index}'
+    params: {
+      name: assignment.name
+      clusterName: redisCluster.name
+      databaseName: redisDatabase.name
+      accessPolicyName: assignment.?accessPolicyName
+      userObjectId: assignment.userObjectId
+    }
+  }
+]
+
+resource database_diagnosticSettings 'Microsoft.Insights/diagnosticSettings@2021-05-01-preview' = [
+  for (diagnosticSetting, index) in (diagnosticSettings ?? []): {
+    name: diagnosticSetting.?name ?? '${name}-diagnosticSettings'
+    properties: {
+      storageAccountId: diagnosticSetting.?storageAccountResourceId
+      workspaceId: diagnosticSetting.?workspaceResourceId
+      eventHubAuthorizationRuleId: diagnosticSetting.?eventHubAuthorizationRuleResourceId
+      eventHubName: diagnosticSetting.?eventHubName
+      logs: [
+        for group in (diagnosticSetting.?logCategoriesAndGroups ?? [{ categoryGroup: 'allLogs' }]): {
+          categoryGroup: group.?categoryGroup
+          category: group.?category
+          enabled: group.?enabled ?? true
+        }
+      ]
+      marketplacePartnerId: diagnosticSetting.?marketplacePartnerResourceId
+      logAnalyticsDestinationType: diagnosticSetting.?logAnalyticsDestinationType
+    }
+    scope: redisDatabase
+  }
+]
+
+module secretsExport 'modules/keyVaultExport.bicep' = if (secretsExportConfiguration != null) {
+  name: '${uniqueString(deployment().name)}-secrets-kv'
+  scope: resourceGroup(
+    split((secretsExportConfiguration.?keyVaultResourceId ?? '//'), '/')[2],
+    split((secretsExportConfiguration.?keyVaultResourceId ?? '////'), '/')[4]
```

```suggestion
    split(secretsExportConfiguration.?keyVaultResourceId!, '/')[2],
    split(secretsExportConfiguration.?keyVaultResourceId!, '/')[4]
```
A recent update to the language allows us to simplify this

---

Change requires ARM template to be rebuild

---

Thanks